### ISSUE TITLE
[Accessibility] Screen Reader Announcing Additional Information

### ISSUE THREAD

**Issue by A-Gattu** [microsoft/fluentui#30665]
### Environment Information

- **Package version(s)**: Details List - Fluent UI 8.57.0
- **Browser and OS versions**: Windows 11, Edge Dev Version 121.0.2277.4 (Official Build) dev (64 - bit)

### Describe the issue:

On the Details List Column Header Screen reader is announcing additional information (<Header_name> section) along with the actual information. The information is redundant and irrelevant and should not be announced.

#### Actual behavior:

For example header name is ‘Status’ and the screen reader announces as ‘Status column header row 1 Status column1, Status section’. 

#### Expected behavior:

For example header name is ‘Status’ and the screen reader should announce it as ‘Status column header row 1 Status column1’. 

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: High

Products/sites affected: All the environments including Production environment.
